### ISSUE TITLE
When draw a round shape, I find an odd phenomenon!

### ISSUE THREAD

**Issue by Jack--Ma** [airbnb/lottie-ios#241]
Hello lottie-iOS developers,
Lottie-iOS is really fantastic, But In using it, with the same JSON file, using in android or web is OK, but when I draw in iOS, I find the circle isn't round...Just like the pic1. Could find the background layer's left and right have some flaws, so as the red circle and the blue circle.
![21b43130-4de6-4ae1-b82f-24ba5489f057](https://cloud.githubusercontent.com/assets/10970854/26283278/b894adcc-3e57-11e7-8a39-ca1c672f43df.png)
I want to find the reason with this odd phenomenon, so I read the source code. I find the layer draw arc path in LOTRectShapeLayer.m with two path, and finally use path2 add to path1, the code is:
```
- (void)_setPath {
/*
 *......
 */
  [path1 appendPath:path2];

  self.path = path1.CGPath;
}
```
And in code LOTEllipseShapeLayer.m, also draw the same path, the code is:
```
- (void)_setPath {
/*
 *......
 */
  // Double path for trim offset.
  [path moveToPoint:circleQ1];
  [path addCurveToPoint:circleQ2 controlPoint1:CGPointMake(circleQ1.x + cpW, circleQ1.y) controlPoint2:CGPointMake(circleQ2.x, circleQ2.y - cpH)];
  
  [path addCurveToPoint:circleQ3 controlPoint1:CGPointMake(circleQ2.x, circleQ2.y + cpH) controlPoint2:CGPointMake(circleQ3.x + cpW, circleQ3.y)];
  
  [path addCurveToPoint:circleQ4 controlPoint1:CGPointMake(circleQ3.x - cpW, circleQ3.y) controlPoint2:CGPointMake(circleQ4.x, circleQ4.y + cpH)];
  
  [path addCurveToPoint:circleQ1 controlPoint1:CGPointMake(circleQ4.x, circleQ4.y - cpH) controlPoint2:CGPointMake(circleQ1.x - cpW, circleQ1.y)];
  
  //Move path(s)
  [path applyTransform:CGAffineTransformMakeTranslation(presentationCircle.circlePosition.x, presentationCircle.circlePosition.y)];
  
  self.path = path.CGPath;
}
```
I feel very strange with these codes, So I try to make the code with:
```
//in first code
//  [path1 appendPath:path2];

//in second code
  // Double path for trim offset.
//  [path moveToPoint:circleQ1];
//  [path addCurveToPoint:circleQ2 controlPoint1:CGPointMake(circleQ1.x + cpW, circleQ1.y) controlPoint2:CGPointMake(circleQ2.x, circleQ2.y - cpH)];
//  
//  [path addCurveToPoint:circleQ3 controlPoint1:CGPointMake(circleQ2.x, circleQ2.y + cpH) controlPoint2:CGPointMake(circleQ3.x + cpW, circleQ3.y)];
//  
//  [path addCurveToPoint:circleQ4 controlPoint1:CGPointMake(circleQ3.x - cpW, circleQ3.y) controlPoint2:CGPointMake(circleQ4.x, circleQ4.y + cpH)];
//  
//  [path addCurveToPoint:circleQ1 controlPoint1:CGPointMake(circleQ4.x, circleQ4.y - cpH) controlPoint2:CGPointMake(circleQ1.x - cpW, circleQ1.y)];
```
And the result is good, like the pic2, could find the round is better than pic1!
![001befe6-44e5-4241-bf25-af1d71a86046](https://cloud.githubusercontent.com/assets/10970854/26283532/2fec768e-3e5d-11e7-858f-0a2542d52168.png)
So...I don't know why lottie-iOS add two same path in a layer? After I changed these codes, Will this operation cause problems?

**Comment by buba447** [comment]
When you say the circle isnt round do you mean that its clipped? Or that the circle is oval in shape? I cant tell from the screenshots. It looks like the circle is slightly clipped.

We have two paths in the layer because each path can have a fill and a stroke, which can be animated individually from one another.

**Comment by buba447** [comment]
Should be fixed in Lottie 2.0! If you still have a problem please create a new issue!
